### PR TITLE
feat(blip-card): new blip-card to show active message template content

### PR DESCRIPTION
**Pull request by vazleal**
Creating a new blip-card to view essential information after sending active messages

### CODE REVIEW

**Review by nakahara-luiz — APPROVED** [review]

**Comment by vazleal** [comment]
Example of the new blip-card:

image

File: src/utils/FormatTextUtils.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+export const constants = {
+  REGEX_FIND_TEXT_BETWEEN_ASTERISKS: /\*([^*]+)\*/g,
+  REGEX_FIND_TEXT_BETWEEN_UNDERSCORES: /_([^*]+)_/g,
+  REGEX_FIND_TEXT_BETWEEN_TILDES: /~([^*]+)~/g,
+  REGEX_FIND_LINE_BREAK: /\\n/g
+}
+
+export const formatText = (text, style) => {
+  console.log('style ', style)
```
Comment (VMarchena) [inline src/utils/FormatTextUtils.js:9]: I think you forgot this console.log

File: src/components/TemplateContent.vue [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+  data: function () {
+    return {
+      alertSvg,
+      alertWhiteSvg,
+      paperplaneSvg
+    }
+  },
+  computed: {
+    showTemplateContentTitle() {
+      return this.unsupportedContentMsg.replace(/_/g, ' ')
```
Comment (VMarchena) [inline src/components/TemplateContent.vue:80]: Does the template content title come from an unsupported content message?

File: src/components/TemplateContent.vue [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+        const links = []
+        attachments.forEach(item => {
+          item.parameters.forEach(parameter => {
+            if (parameter.type === 'image') {
+              links.push(parameter.image.link)
+            } else if (parameter.type === 'video') {
+              links.push(parameter.video.link)
+            } else if (parameter.type === 'audio') {
+              links.push(parameter.audio.link)
+            }
```
Comment (VMarchena) [inline src/components/TemplateContent.vue:120]:
```suggestion
           const { type } = parameter
            if (['image', 'video', 'audio'].includes(type)) {
              links.push(parameter[type].link)
            } 
```

I don't know if you can remove that second `forEach` but I think you can simplify this `if` like this.  What do you think?

File: src/components/TemplateContent.vue [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+        <img v-if="this.status === 'waiting'" :src="clockSvg">
+        <img v-else-if="this.status === 'accepted'" :src="checkSentSvg">
+        <img v-else-if="this.status === 'received'" :src="doubleCheckReceivedSvg">
+        <img v-else-if="this.status === 'consumed'" :src="doubleCheckReadSvg">
+        <div v-else-if="this.status === 'failed'" class="failure">
+          {{ failedToSendMsg }}
+        </div>
+      </span>
+      <div>{{ date }}</div>
+    </div>
```
Comment (VMarchena) [inline src/components/TemplateContent.vue:32]: I think you can replace this with the `BlipCardDate` component.

File: src/utils/FormatTextUtils.js (hunk 1 above)
Comment (vazleal) [reply src/utils/FormatTextUtils.js]: Thanks!

File: src/components/TemplateContent.vue (hunk 3 above)
Comment (vazleal) [reply src/components/TemplateContent.vue]: Yes, I agree on the simplification, I did not know something like this was possible in JS, thanks for the suggestion.

And about forEach, I tried to remove it but the implicit logic in the object blocks this improvement

File: src/components/TemplateContent.vue (hunk 4 above)
Comment (vazleal) [reply src/components/TemplateContent.vue]: Thanks for the suggestion!

**Review by VMarchena — APPROVED** [review]

**Review by ceciliamariar — APPROVED** [review]

**Review by DanielCouto — APPROVED** [review]

**Pull request merged**